core/src/main/scala/kafka/utils/ShutdownableThread.scala

```diff
@@ -57,15 +57,14 @@ abstract class ShutdownableThread(val name: String, val isInterruptible: Boolean
   def doWork(): Unit
 
   override def run(): Unit = {
-    info("Starting ")
+    info("Starting %s %d".format(name, this.getId))
     try{
       while(isRunning.get()){
         doWork()
       }
     } catch{
       case e: Throwable =>
-        if(isRunning.get())
-          error("Error due to ", e)
+        error("Error due to ", e)
```

Comment:
I think the check on if(isRunning.get()) is still useful. For example, if the thread is interruptible and we are shutting down the thread, we don't need to log error due to InterruptedException.